server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java

```diff
@@ -276,19 +259,17 @@ public void testCancelFailedSearchWhenPartialResultDisallowed() throws Exception
             queryLatch.countDown();
             assertBusy(() -> {
                 final List<SearchTask> searchTasks = getSearchTasks();
-                // The search request can complete before the "cancelledLatch" is latched if the second shard request is sent
-                // after the request was cancelled (i.e., the child task is not allowed to start after the parent was cancelled).
                 if (searchTasks.isEmpty() == false) {
                     assertThat(searchTasks, hasSize(1));
                     assertTrue(searchTasks.get(0).isCancelled());
                 }
             }, 30, TimeUnit.SECONDS);
         } finally {
+            searchThread.join();
```

Comment:
By waiting for the search thread to finish, we no longer (strictly) need the `assertBusy(() -> {` block at L#279 before cleaning up the plugins.

However, I decided to keep it because I like that the line 
```java
 assertTrue(searchTasks.get(0).isCancelled());
 ```
 gives some extra assurance by checking that the that the task was in fact cancelled.